### PR TITLE
Set client timestamp on outgoing analytics events

### PR DESCRIPTION
**Pull request by dvirsky**
This PR adds a timestamp of the current client machine to every outgoing analytics request.

### CODE REVIEW

**Comment by dvirsky** [comment]
The checks are failing from something unrelated to my PR :/

File: src/runtime/analytics-service.js [hunk 1]
```diff
@@ -130,6 +131,13 @@ export class AnalyticsService {
     // the user wait too long.
     /** @private {?number} */
     this.timeout_ = null;
+
+    // Time source for setting the client timestamp before sending requests.
+    // Injectable for tests.
+    /** @private {function():!../proto/api_messages.Timestamp} */
+    this.timeSource_ = () => {
```
Comment (ChrisAntaki) [inline src/runtime/analytics-service.js:138]: Could you call this `getTimestamp_` ? I think that's clearer than `timeSource_`

File: src/runtime/analytics-service.js [hunk 2]
```diff
@@ -150,6 +158,14 @@ export class AnalyticsService {
     }
   }
 
+  /**
```
Comment (ChrisAntaki) [inline src/runtime/analytics-service.js:161]: It's simpler for tests to just override the `getTimestamp_` property

File: src/runtime/analytics-service.js [hunk 3]
```diff
@@ -315,7 +331,7 @@ export class AnalyticsService {
     const meta = new AnalyticsEventMeta();
     meta.setEventOriginator(event.eventOriginator);
     meta.setIsFromUserAction(!!event.isFromUserAction);
-
+    this.updateContextTimestamp_();
```
Comment (ChrisAntaki) [inline src/runtime/analytics-service.js:334]: Since it's called just once, and it's a one liner, could you inline it for simplicity ?

Comment (dvirsky) [reply src/runtime/analytics-service.js:334]: Done.

File: src/runtime/analytics-service.js (hunk 1 above)
Comment (dvirsky) [reply src/runtime/analytics-service.js:138]: Done.

File: src/runtime/analytics-service.js (hunk 2 above)
Comment (dvirsky) [reply src/runtime/analytics-service.js:161]: Done. Old habbits from C++ :)

**Review by ChrisAntaki — APPROVED** [review]

**Pull request merged**